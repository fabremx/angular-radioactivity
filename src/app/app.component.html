<div>
  <h1 class='title'>{{ title }}</h1>

  <div *ngFor="let item of items" class="item">
    <div class="item__name">{{ item.name }} ({{ item.zipcode }})</div>
    <div ngClass="item__legend {{ getRadioActivityLevel(item.value) }}"></div>
  </div>
</div>