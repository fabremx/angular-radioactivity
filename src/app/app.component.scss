.item {
    display: flex;
    margin-top: 10px;
}

.item__name {
    width: 100px;
}
.item__legend {
    width: 20px;
    height: 20px;
    margin-left: 10px;
    border: 1px solid black;
}
.item__legend.low { background-color: #E0E080 }
.item__legend.medium { background-color: #E0E040 }
.item__legend.high { background-color: #E08000 }
.item__legend.critical { background-color: #FF0000 }