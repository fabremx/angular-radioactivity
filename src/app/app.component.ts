import { Component } from '@angular/core';
import { RadioactivityService, RadioactivityLevel } from './radioactivity.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  constructor(
    private radioactivityService: RadioactivityService,
  ) { }

  title = 'Radioactrivity';
  items = [
    { name: 'Paris', zipcode: '75', value: 52 },
    { name: 'Bordeaux', zipcode: '33', value: 120 },
    { name: 'Lyon', zipcode: '69', value: 28 },
    { name: 'Lille', zipcode: '59', value: 251 },
    { name: 'Marseille', zipcode: '13', value: 78 },
    { name: 'Nantes', zipcode: '44', value: 102 }
  ]

  getRadioActivityLevel(value: number): RadioactivityLevel {
    return this.radioactivityService.getRadioacvityLevel(value)
  }
}
