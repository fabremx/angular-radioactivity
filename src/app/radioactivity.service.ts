import { Injectable } from '@angular/core';

export type RadioactivityLevel = 'low' | 'medium' | 'high' | 'critical';
interface MappingRadioactivityLevel {
  start: number
  end: number
  output: RadioactivityLevel
}

@Injectable({
  providedIn: 'root'
})
export class RadioactivityService {

  constructor() { }

  getRadioacvityLevel(radioactivityValue: number): RadioactivityLevel {
    const mapping: MappingRadioactivityLevel[] = [
      { start: 0, end: 50, output: 'low' },
      { start: 51, end: 100, output: 'medium' },
      { start: 101, end: 150, output: 'high' },
      { start: 151, end: Infinity, output: 'critical' },
    ]

    return (mapping
      .find((map) => this.isRange(radioactivityValue, map.start, map.end)) as MappingRadioactivityLevel)
      .output
  }

  private isRange(value: number, start: number, end: number): boolean {
    return value >= start && value <= end;
  }
}
