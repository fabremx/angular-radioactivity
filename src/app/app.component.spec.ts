import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  it('should render correctly the component', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
