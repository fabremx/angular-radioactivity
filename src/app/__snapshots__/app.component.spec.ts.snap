// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render correctly the 1`] = `
<app-root
  items={[Function Array]}
  radioactivityService={[Function RadioactivityService]}
  title={[Function String]}
>
  <div>
    <h1
      class="title"
    >
      Radioactrivity
    </h1>
    
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Paris (75)
      </div>
      <div
        class="item__legend medium"
      />
    </div>
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Bordeaux (33)
      </div>
      <div
        class="item__legend high"
      />
    </div>
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Lyon (69)
      </div>
      <div
        class="item__legend low"
      />
    </div>
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Lille (59)
      </div>
      <div
        class="item__legend critical"
      />
    </div>
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Marseille (13)
      </div>
      <div
        class="item__legend medium"
      />
    </div>
    <div
      class="item"
    >
      <div
        class="item__name"
      >
        Nantes (44)
      </div>
      <div
        class="item__legend high"
      />
    </div>
  </div>
</app-root>
`;
